content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,501 @@
+        return
+    }
+    
+    res, err := client.Ping(context.Background())
+    if err != nil {
+        fmt.Println("There was an error: ", err)
+        return
+    }
+    fmt.Println(res) // PONG
+    client.Close()
+}
+```
+
+## Advanced Configuration Options
+
+### Read Strategies for Optimized Performance
+
+Balance consistency and throughput with flexible read strategies:
+
+```go
+// Configure to prefer replicas for read operations
+cfg := config.NewClusterClientConfiguration().
+    WithAddress(&config.NodeAddress{Host: "cluster.example.com", Port: 6379}).
+    WithReadFrom(config.PreferReplica)
+
+client, err := glide.NewClusterClient(cfg)
+
+// Write to primary
+client.Set(context.Background(), "key1", "value1")
+
+// Automatically reads from a replica (round-robin)
+result, err := client.Get(context.Background(), "key1")
+```
+
+Available strategies:
+
+- **Primary**: Always read from primary nodes for the freshest data
+- **PreferReplica**: Distribute reads across replicas in round-robin fashion, falling back to primary when needed
+- **AzAffinity**: Prefer replicas in the same availability zone as the client
+- **AzAffinityReplicaAndPrimary**: Spread the read requests among nodes within the client's availability zone in a round-robin manner
+
+### Authentication and TLS
+
+Secure your connections with built-in authentication and TLS support:
+
+```go
+// Configure with authentication
+cfg := config.NewClientConfiguration().
+    WithAddress(&config.NodeAddress{Host: "localhost", Port: 6379}).
+    WithCredentials(config.NewServerCredentials("username", "password")).
+    WithUseTLS(true) // Enable TLS for encrypted connections
+```
+
+### Request Timeout and Handling
+
+Fine-tune timeout settings for different workloads:
+
+```go
+// Set a longer timeout for operations that may take more time
+cfg := config.NewClientConfiguration().
+    WithAddress(&config.NodeAddress{Host: "localhost", Port: 6379}).
+    WithRequestTimeout(500) // 500ms timeout
+```
+
+## Behind the Scenes: Technical Architecture
+
+The Valkey GLIDE Go client is built on top of the Valkey GLIDE core. The core framework is written in Rust (lib.rs), which exposes public functions. These functions are converted to a C header file using Cbindgen. The Go client then uses CGO to call these C functions, providing Go developers with an idiomatic interface while leveraging Rust's performance advantages. This architecture ensures consistent behavior across all Valkey GLIDE language implementations (Java, Python, Node.js, and Go) while maintaining performance and reliability.
+
+### Component details
+
+```text
++------------+      +------+      +------------+      +------------+      +------------+      +------------+
+|            |      |      |      |            |      |            |      |            |      |            |
+|    Go      |----->|      |----->|  C Header  |----->|    FFI     |----->|    Rust    |----->|   Valkey   |
```

Comment:
I'd remove CGO and cbindgen from there. They are used on build phase, but not in runtime. Well, cgo still there, but it is a part of go.

Reply:
Agreed.  Will make this change